testapps/test_pillowtop/tests/test_domain_pillow.py

```diff
@@ -65,6 +66,44 @@ def test_kafka_domain_pillow_deletions(self):
         # ensure removed from ES
         self.assertEqual(0, DomainES().run().total)
 
+    def test_reverted_domain_pillow_deletion(self):
+        domain_name = 'domain-pillow-delete'
+        with drop_connected_signals(commcare_domain_post_save):
+            domain = create_domain(domain_name)
+
+        # send to kafka
+        since = get_topic_offset(topics.DOMAIN)
+        publish_domain_saved(domain)
+
+        # send to elasticsearch
+        pillow = get_domain_kafka_to_elasticsearch_pillow()
+        pillow.process_changes(since=since, forever=False)
+        self.elasticsearch.indices.refresh(self.index_info.index)
+
+        # verify there
+        self._verify_domain_in_es(domain_name)
+
+        domain_obj = Domain.get_by_name('domain-pillow-delete')
+        domain_obj.doc_type = 'Domain-DUPLICATE'
+        domain_obj.save()
```

Comment:
Is this considered a _delete_ operation? Otherwise I'm not seeing how the "undelete" below is doing that.

Reply:
This is a change that triggers an ES deletion (https://github.com/dimagi/commcare-hq/blob/74bc31910f692126f03c46a350ab8ae5700f87dd/corehq/apps/change_feed/document_types.py#L58) and is how the delete test above this one triggers that, so I used it for consistency.